### PR TITLE
Doc: update etcd configuration detail with config file

### LINKED ISSUE #9063: docs: include mention of etcd configuration yaml in configuration.md
[Configuration.md](https://github.com/coreos/etcd/blob/master/Documentation/op-guide/configuration.md) doesn't mention the [etcd.conf file](https://github.com/coreos/etcd/blob/master/etcd.conf.yml.sample) format or the order of precedent between flags, file, and environment variables.

### PR DESCRIPTION
**Pull request by spzala**
Add a reference of the config file, sample file link and precedence detail  to the
configuration doc.

Fixes #9063 

# Contributing guidelines

Please read our [contribution workflow][contributing] before submitting a pull request.

[contributing]: https://github.com/coreos/etcd/blob/master/CONTRIBUTING.md#contribution-flow

### CODE REVIEW

**Comment by spzala** [comment]
/cc @gyuho @xiang90  
Since the config file itself is used through a command line flag, I didn't create a new section but tried to add detail in the current sections on configuration flags. Thanks!

**Comment by xiang90** [comment]
Do we have a doc about the syntax and options for the config file? For
example how do we map command line flags to config file yaml fields?

**Comment by spzala** [comment]
@xiang90 it seems like configuration.md is the doc which has all the flags usage and syntax documented that can be used in the config file.

**Comment by spzala** [comment]
@xiang90 I have tried modifying a line to show the relationship/mapping between file and flags as "A reusable configuration file is a YAML file made of one or more command-line flags...." if that look good? Thanks!

File: Documentation/op-guide/configuration.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Configuration flags
 
-etcd is configurable through command-line flags and environment variables. Options set on the command line take precedence over those from the environment.
+etcd is configurable through a configuration file, various command-line flags, and environment variables.
+
+A reusable configuration file is a YAML file made of one or more command-line flags described below. In order to use this file, specify the file path as a value to the `--config-file` flag. The [sample configuration file][sample-config-file] can be used as a starting point to create a new configuration file as needed.
+
+Options set on the command line take precedence over those from the environment. If a configuration file is provided, other command line flags will be ignored.
```
Comment (xiang90) [inline Documentation/op-guide/configuration.md]: If a configuration file is provided, will env also be ignored? or they will be overwritten?

Comment (spzala) [reply Documentation/op-guide/configuration.md]: @xiang90 env variables will be ignored in that case.

Comment (xiang90) [reply Documentation/op-guide/configuration.md]: can we clarify that here too besides `other command line flags will be ignored`?

Comment (spzala) [reply Documentation/op-guide/configuration.md]: Sure

Comment (xiang90) [reply Documentation/op-guide/configuration.md]: probably we should also add some logging in etcd itself saying `config file is provided, xxx are ignored.`

Comment (xiang90) [reply Documentation/op-guide/configuration.md]: no need to be in this PR though.

Comment (spzala) [reply Documentation/op-guide/configuration.md]: Agree, both suggestions makes sense!

Comment (spzala) [reply Documentation/op-guide/configuration.md]: I have created second commit with your suggestions related to ignoring flags. Thanks!

File: Documentation/op-guide/configuration.md [hunk 2]
```diff
@@ -2,9 +2,10 @@
 
 etcd is configurable through a configuration file, various command-line flags, and environment variables.
 
-A reusable configuration file is a YAML file made of one or more command-line flags described below. In order to use this file, specify the file path as a value to the `--config-file` flag. The [sample configuration file][sample-config-file] can be used as a starting point to create a new configuration file as needed.
+A reusable configuration file is a YAML file made with name and value of one or more command-line flags described below. In order to use this file, specify the file path as a value to the `--config-file` flag. The [sample configuration file][sample-config-file] can be used as a starting point to create a new configuration file as needed.
 
 Options set on the command line take precedence over those from the environment. If a configuration file is provided, other command line flags will be ignored.
```
Comment (xiang90) [inline Documentation/op-guide/configuration.md]: If a configuration file is provided, other command line flags will be ignored. 

-> 

If a configuration file is provided, other command line flags and env variables will be ignored.

**Comment by xiang90** [comment]
LGTM after add `env variable` clarification, and squash the commits into one. Thanks!

**Comment by spzala** [comment]
@xiang90 sorry, squashing was messaged up so I had to push few times to fix it but I hope it looks good now. Thanks!

**Comment by xiang90** [comment]
Thanks. Will merge when CI is green. Merry Christmas.

**Comment by spzala** [comment]
@xiang90 thanks!! Merry Christmas.

**Pull request merged**